### PR TITLE
fix(rxSearchbox): FRMW-714 support undefined and null value

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: https://jira.rax.io/browse/FRMW-714

### LGTMs
- [x] Dev LGTM
- [x] Dev LGTM
- [x] QE LGTM

### CODE REVIEW

File: src/components/rxSearchBox/scripts/rxSearchBox.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ angular.module('encore.ui.rxSearchBox')
             $scope.rxPlaceholder = $scope.rxPlaceholder || 'Search...';
 
             $scope.$watch('searchVal', function (newVal) {
-                if ($scope.isDisabled) {
+                if (!newVal || $scope.isDisabled) {
                     $scope.isClearable = false;
                 } else {
                     $scope.isClearable = newVal.toString() !== '';
```
Comment (Droogans) [inline src/components/rxSearchBox/scripts/rxSearchBox.js]: This ought to be a function.

`</pedantry>`

Comment (CITguy) [inline src/components/rxSearchBox/scripts/rxSearchBox.js]: Which part? left or right side of assignment?

Comment (Droogans) [inline src/components/rxSearchBox/scripts/rxSearchBox.js]:
```js
$scope.isClearable = function () {
    return newVal !== undefined && newVal.toString() !== '';
}
```

I'm shooting off  the hip here -- that may be the wrong inner definition for `$scope.isClearable()` the function.

(this has nothing to do with your PR, I just noticed it.)

**Comment by parlarjb** [comment]
LGTM

File: src/components/rxSearchBox/scripts/rxSearchBox.spec.js [hunk 2]
```diff
@@ -14,73 +14,131 @@ describe('rxSearchBox', function () {
...
                 beforeEach(function () {
-                    scope.clearSearch();
+                    scope.testModel = 'hiyoo';
+                    scope.$apply();
                 });
 
-                it('should set searchVal to empty string', function () {
-                    expect(scope.searchVal).to.be.empty;
+                it('should not be clearable', function () {
+                    expect(isoScope.isClearable).to.be.true;
```
Comment (100stacks) [inline src/components/rxSearchBox/scripts/rxSearchBox.spec.js]: Based on your `it` clause, should the `expect` clause be `false`?

**Comment by 100stacks** [comment]
Other than the one question...LGTM

File: src/components/rxSearchBox/scripts/rxSearchBox.spec.js (hunk 2 above)
Comment (CITguy) [inline src/components/rxSearchBox/scripts/rxSearchBox.spec.js]: good catch, description is incorrect. will update

**Pull request merged**